File: Problem002-Add Two Numbers/Solution.java
/*
 * @SEU(master);@ZZULI(bachelor)
 * author:CuiXuan
 * email:873059043@qq.com
 * If you find any errors, please let me know.
 * If you have any better solution, please let me know.
 * 
 * Solution Using time 472ms
 */

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int high=(l1.val+l2.val)/10;
        ListNode head = new ListNode((l1.val+l2.val)%10);
        ListNode p=head;
        l1=l1.next;l2=l2.next;
        while(l1!=null|| l2!=null){
            int a = l1==null?0:l1.val;
            int b = l2==null?0:l2.val;
            ListNode s=new ListNode((a+b+high)%10);
            high = (a+b+high)/10;
            p.next=s;
            p=s;
            l1 = l1==null?null:l1.next;
            l2 = l2==null?null:l2.next;
        }
        if(high!=0){
            ListNode s=new ListNode(high);
            p.next=s;
            p=s;
        }
        return head;
    }
}

File: Problem002-Add Two Numbers/Solution.py
"""
 @SEU(master);@ZZULI(bachelor)
 author:CuiXuan
 email:873059043@qq.com
 If you find any errors, please let me know.
 If you have any better solution, please let me know.

 Solution Using time 228ms
"""


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} l1
    # @param {ListNode} l2
    # @return {ListNode}
    def addTwoNumbers(self, l1, l2):
        high = (l1.val+l2.val)/10;
        head = ListNode((l1.val+l2.val)%10)
        p=head;
        l1=l1.next
        l2=l2.next
        while l1!=None or l2!=None:
            a =0 if l1==None else l1.val
            b =0 if l2==None else l2.val
            s = ListNode((a+b+high)%10)
            s.next=None
            p.next=s
            p=s
            high = (a+b+high)/10
            l1 =None if l1==None else l1.next
            l2 =None if l2==None else l2.next
        if high!=0:
            s = ListNode(high)
            s.next=None
            p.next=s
        return head